### PR TITLE
LIVE-2788: use params to override theme

### PR DESCRIPTION
**Pull request by joecowton1**
## Why are you doing this?

We need to override article theme styles based on which fronts container they live in. From the perspective of Editions Rendering this means we need to be able to pass a parameter containing the `theme` value from Editions backend, and use that to override the existing article theme value.

## Changes

- get `themeOverride` from query params
- pass value through to article `item` level

### CODE REVIEW

File: src/server/server.ts [hunk 1]
```diff
@@ -205,11 +214,12 @@ async function serveEditionsArticlePost(
 	try {
 		// The "req.body" should contain a 'Content' object which fetched by the
 		// Edition backend from the capi
+		const themeOverride = req.query.theme as Theme | undefined;
```
Comment (joecowton1) [inline src/server/server.ts:217]: @JamieB-gu how do I parse this so i'm not type casting please?

File: src/server/server.ts [hunk 2]
```diff
@@ -185,11 +193,12 @@ async function serveArticlePost(
 		const renderingRequest = await mapiDecoder(req.body);
 		const richLinkDetails = req.query.richlink === '';
 		const isEditions = req.query.editions === '';
+		const themeOverride = req.query.theme as Theme | undefined;
```
Comment (joecowton1) [inline src/server/server.ts:196]: Same here.

File: src/server/server.ts [hunk 3]
```diff
@@ -244,11 +254,17 @@ async function serveArticleGet(
 				};
 
 				const richLinkDetails = req.query.richlink === '';
+				const themeOverride = req.query.theme as Theme | undefined;
```
Comment (joecowton1) [inline src/server/server.ts:257]: And here..

File: src/server/server.ts [hunk 4]
```diff
@@ -185,11 +245,17 @@ async function serveArticlePost(
...
 		if (richLinkDetails) {
 			void serveRichLinkDetails(renderingRequest, res);
 		} else {
-			void serveArticle(renderingRequest, res, isEditions);
+			void serveArticleSwitch(
+				renderingRequest,
+				res,
+				isEditions,
+				themeOverride,
+			);
```
Comment (JamieB-gu) [inline src/server/server.ts:258]: Is this needed in `serveArticlePost`? This function is never triggered for Editions is it? Editions uses `serveEditionsArticlePost`?

Comment (joecowton1) [reply src/server/server.ts]: Good point, fixed!

**Review by JamieB-gu — APPROVED** [review]

**Pull request merged**